Source/AliveLibAO/Bullet.cpp

```diff
@@ -35,4 +43,196 @@ BaseGameObject* Bullet::VDestructor(signed int flags)
 }
 
+EXPORT void Bullet::VUpdate_408E30()
+{
+    if (!gMap_507BA8.Is_Point_In_Current_Camera_4449C0(
+        field_28_level,
+        field_2A_path,
+        field_18_xpos,
+        field_1C_ypos,
+        0
+        ))
+    {
+        field_6_flags.Set(BaseGameObject::eDead_Bit3);
+        return;
+    }
+    const short volume = field_2C_scale != FP_FromDouble(0.5) ? 75 : 50;
+
+    switch (field_10_type)
+    {
+        case BulletType::Type_0:
+        case BulletType::Type_1:
+        {
+            FP hitX = {};
+            FP hitY = {};
+            PSX_RECT shootRect = {};
+            if (field_20 > FP_FromInteger(0))
+            {
+                shootRect.x = FP_GetExponent(field_18_xpos);
+                shootRect.w = FP_GetExponent(field_18_xpos) + 640;
+            }
+            else
+            {
+                shootRect.x = FP_GetExponent(field_18_xpos) - 640;
+                shootRect.w = FP_GetExponent(field_18_xpos);
+            }
+            shootRect.y = FP_GetExponent(field_1C_ypos - FP_FromInteger(15));
+            shootRect.h = FP_GetExponent(field_1C_ypos + FP_FromInteger(5));
+            BaseAliveGameObject *pShotObj = ShootObject_409400(&shootRect);
+            if (sCollisions_DArray_504C6C->RayCast_40C410(
+                    field_18_xpos,
+                    field_1C_ypos,
+                    field_20 + field_18_xpos,
+                    field_1C_ypos,
+                    &field_14_pLine,
+                    &hitX,
+                    &hitY,
+                    field_2C_scale != FP_FromDouble(0.5) ? 7 : 0x70
+                ))
+            {
+                FP distHit = {};
+                FP distShot = {};
+                if (pShotObj)
+                {
+                    distHit = FP_Abs(hitX - field_18_xpos);
+                    distShot = FP_Abs(pShotObj->field_A8_xpos - field_18_xpos);
+                }
+                if (!pShotObj || (distShot > distHit))
+                {
+                    if (field_20 <= FP_FromInteger(0))
+                    {
+                        auto pSpark = ao_new<Spark>();
+                        if (pSpark)
+                        {
+                            pSpark->ctor_477B70(hitX, hitY, field_2C_scale, 6u, -76, 76);
+                        }
+                    }
+                    else
+                    {
+                        auto pSpark = ao_new<Spark>();
+                        if (pSpark)
+                        {
+                            pSpark->ctor_477B70(hitX, hitY, field_2C_scale, 6u, 50, 205);
+                        }
+                    }
+                    New_Particles_419A80(hitX, hitY, field_2C_scale, 3, 0);
+                    if (Math_RandomRange_450F20(0, 100) >= 90)
+                    {
+                        field_6_flags.Set(BaseGameObject::eDead_Bit3);
+                        return;
+                    }
+                    if (Math_RandomRange_450F20(0, 128) < 64)
```

Comment:
this can be another if/else chain with the common  field_6_flags.Set(BaseGameObject::eDead_Bit3); exit/return path